lib/minheap.js

```diff
@@ -11,18 +11,57 @@ class MinHeap {
   }
 
   // This method adds a HeapNode instance to the heap
-  // Time Complexity: ?
-  // Space Complexity: ?
+  // Time Complexity: o(log n)because the while loop where we only switch with the parent only deals with half the nodes at most.
+  // Space Complexity: o(1) we are just adding a node.
   add(key, value = key) {
```

Comment:
You seem to be using `heapUp` as a swap method.  This works, but I think it makes the `add` method a little more complicated than it needs to be.